docs/user/azure/install_upi_azurestack.md

```diff
@@ -0,0 +1,604 @@
+
+Now we can create the bootstrap ignition configs:
+
+```console
+$ openshift-install create ignition-configs
+INFO Consuming Openshift Manifests from target directory
+INFO Consuming Worker Machines from target directory
+INFO Consuming Common Manifests from target directory
+INFO Consuming Master Machines from target directory
+```
+
+After running the command, several files will be available in the directory.
+
+```console
+$ tree
+.
+├── auth
+│   └── kubeconfig
+├── bootstrap.ign
+├── master.ign
+├── metadata.json
+└── worker.ign
+```
+
+## Create The Resource Group
+
+Use the command below to create the resource group in the selected Azure region:
+
+```sh
+az group create --name $RESOURCE_GROUP --location $AZURE_REGION
+```
+
+## Upload the files to a Storage Account
+
+The deployment steps will read the Red Hat Enterprise Linux CoreOS virtual hard disk (VHD) image and the bootstrap ignition config file
+from a blob. Create a storage account that will be used to store them and export its key as an environment variable.
+
+```sh
+az storage account create -g $RESOURCE_GROUP --location $AZURE_REGION --name ${CLUSTER_NAME}sa --kind Storage --sku Standard_LRS
+export ACCOUNT_KEY=`az storage account keys list -g $RESOURCE_GROUP --account-name ${CLUSTER_NAME}sa --query "[0].value" -o tsv`
+```
+
+### Copy the cluster image
+
+In order to create VMs, the RHCOS VHD must be available in the Azure Stack
+environment. The VHD should be downloaded, decompressed, and uploaded to a
+storage blob.
+
+First, download and decompress the VHD, note that the decompressed file is 16GB:
+
+```sh
+$ export COMPRESSED_VHD_URL=$(curl -s https://raw.githubusercontent.com/openshift/installer/release-4.9/data/data/rhcos-amd64.json | jq -r '(.baseURI + .images.azurestack.path)')
+$ wget $COMPRESSED_VHD_URL
+--2021-08-05 14:40:05--  https://rhcos-redirector.apps.art.xq1c.p1.openshiftapps.com/art/storage/releases/rhcos-4.9/49.84.202107010027-0/x86_64/rhcos-49.84.202107010027-0-azurestack.x86_64.vhd.gz
+Resolving rhcos-redirector.apps.art.xq1c.p1.openshiftapps.com (rhcos-redirector.apps.art.xq1c.p1.openshiftapps.com)... 3.229.121.196, 3.232.64.216, 3.225.72.82, ...
+Connecting to rhcos-redirector.apps.art.xq1c.p1.openshiftapps.com (rhcos-redirector.apps.art.xq1c.p1.openshiftapps.com)|3.229.121.196|:443... connected.
+HTTP request sent, awaiting response... 200 OK
+Length: 1030872938 (983M) [application/gzip]
+Saving to: ‘rhcos-49.84.202107010027-0-azurestack.x86_64.vhd.gz’
+$ gunzip rhcos-49.84.202107010027-0-azurestack.x86_64.vhd.gz
+```
+
+Next, create a container for the VHD:
+
+```sh
+az storage container create --name vhd --account-name ${CLUSTER_NAME}sa
+```
+
+As mentioned above, the VHD size is massive. If you have a fast upload speed,
+you may be satisfied to upload the whole 16GB file:
+
+```sh
+az storage blob upload --account-name ${CLUSTER_NAME}sa --account-key $ACCOUNT_KEY -c vhd -n "rhcos.vhd" -f rhcos-49.84.202107010027-0-azurestack.x86_64.vhd
```

Comment:
Not sure but can this help to make it faster? Instead of downloading and uploading, directly passing a url for upload.
https://docs.microsoft.com/en-us/cli/azure/storage/blob?view=azure-cli-latest#az_storage_blob_rewrite

Reply:
Nvm need to extract it first.